x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SetSingleNodeAllocateStep.java

```diff
@@ -45,18 +45,37 @@
     private static final Logger logger = LogManager.getLogger(SetSingleNodeAllocateStep.class);
     public static final String NAME = "set-single-node-allocation";
 
-    public SetSingleNodeAllocateStep(StepKey key, StepKey nextStepKey, Client client) {
+    private Integer numberOfShards;
+
+    public SetSingleNodeAllocateStep(StepKey key, StepKey nextStepKey, Client client, Integer numberOfShards) {
         super(key, nextStepKey, client);
+        this.numberOfShards = numberOfShards;
     }
 
     @Override
     public boolean isRetryable() {
         return true;
     }
 
+    public Integer getNumberOfShards() {
+        return numberOfShards;
+    }
+
     @Override
     public void performAction(IndexMetadata indexMetadata, ClusterState clusterState,
                               ClusterStateObserver observer, ActionListener<Boolean> listener) {
+        String indexName = indexMetadata.getIndex().getName();
+        if (numberOfShards != null) {
+            int sourceNumberOfShards = indexMetadata.getNumberOfShards();
+            int factor = sourceNumberOfShards / numberOfShards;
+            if (factor * numberOfShards < sourceNumberOfShards) {
```

Comment:
Just curious, is there any particular reason you didn't go with `if (sourceNumberOfShards % numberOfShards != 0) {`?

Reply:
Just a habit, `%` is better, I think.